### PR TITLE
LibPDF+pdf: Enough stuff to be able to read password-protected PDFs created by macOS

### PR DESCRIPTION
**Pull request by nico**
In particular, implements support for AES in LibPDF.

Also fixes an issue where the linearization detection logic would crash on inputs where the first object is an encrypted compressed stream in a not-linearized file.

### CODE REVIEW

**Comment by nico** [comment]
@mattco98 for LibPDF.

@alimpfard for the AES client code. I found the API a bit awkward to use: From what I understand, I have to use the span that `encrypt()` / `decrypt()` can modify to access the encrypted / decrypted data, which means I can't move the ByteBuffer I have elsewhere but have to make a copy of just the bytes in the span (?)

File: Userland/Libraries/LibPDF/Encryption.cpp [hunk 1]
```diff
@@ -57,6 +60,56 @@ PDFErrorOr<NonnullRefPtr<SecurityHandler>> SecurityHandler::create(Document* doc
     TODO();
 }
 
+static PDFErrorOr<void> parse_v4_crypt(Document* document, NonnullRefPtr<DictObject> encryption_dict, DeprecatedString filter, CryptFilterMethod& method, size_t& length_in_bits)
```
Comment (mattco98) [inline Userland/Libraries/LibPDF/Encryption.cpp:63]: Feels like a simple struct return value would be nicer than two out-params

**Comment by nico** [comment]
My tab here had a comment about giving `parse_v4_crypt` a return type instead of two out params. I've done that, but now that I try to reply to that comment it tells me the comment's been deleted. LMK if you'd prefer me to change it back to the out params. I like the struct return :)

**Comment by mattco98** [comment]
> My tab here had a comment about giving parse_v4_crypt a return type instead of two out params. I've done that, but now that I try to reply to that comment it tells me the comment's been deleted. LMK if you'd prefer me to change it back to the out params. I like the struct return :)

Yeah idk what GitHub did there, I started the review and then _thought_ I hit cancel, but I guess I didn't. Anyways, I deleted it because the values coming in had some default values, but it is probably better this way regardless 😄

**Review by mattco98 — APPROVED** [review]

**Pull request merged**